src/Native/Unix/System.Net.Http.Native/pal_multi.cpp

```diff
@@ -69,8 +70,31 @@ extern "C" int32_t HttpNative_MultiWait(CURLM* multiHandle,
     int numFds;
     CURLMcode result = curl_multi_wait(multiHandle, &extraFds, 1, FailsafeTimeoutMilliseconds, &numFds);
 
-    *isExtraFileDescriptorActive = (extraFds.revents & CURL_WAIT_POLLIN) != 0;
-    *isTimeout = numFds == 0;
+    if (numFds == 0)
+    {
+        *isTimeout = true;
+        *isExtraFileDescriptorActive = false;
+    }
+    else
+    {
+        *isTimeout = false;
+
+        //
+        // Prior to libcurl version 7.32.0, the revents field was not returned properly for "extra" file descriptors
+        // passed to curl_multi_wait.  See https://github.com/dotnet/corefx/issues/9751.  So if we have a libcurl
+        // prior to that version, we need to do our own poll to get the status of the extra file descriptor.
+        //
+        if (curl_version_info(CURLVERSION_NOW)->version_num >= 0x073200)
```

Comment:
I originally wrote the code to cache the `curl_version_info` result, but it turns out that `curl_version_info` already does that internally.  Calling this should be very cheap.